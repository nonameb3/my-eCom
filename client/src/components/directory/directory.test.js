import React from 'react';
import { shallow } from 'enzyme';

import Directory from './directory.component';
import MenuItem from "../menu-item/menu-item.component";

let wrapper;
const mockCollection = [
  {
    title: "hats",
    imageUrl: "https://i.ibb.co/cvpntL1/hats.png",
    id: 1,
    linkUrl: "shop/hats"
  },
  {
    title: "jackets",
    imageUrl: "https://i.ibb.co/px2tCc3/jackets.png",
    id: 2,
    linkUrl: "shop/jackets"
  }
];

jest.mock("react-redux", () => ({
  useSelector: jest.fn(() => mockCollection)
}));

describe('directory component', () => {
  beforeEach(() => {
    wrapper = shallow(<Directory />);
  });

  it('should render directory component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('sould render munu-item as expect', () => {
    expect(wrapper.find(MenuItem).length).toBe(mockCollection.length);
  })
})