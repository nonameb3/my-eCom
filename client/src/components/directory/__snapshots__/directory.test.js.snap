// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directory component should render directory component 1`] = `ShallowWrapper {}`;
